### DISCUSSION TITLE
The Nordigen API is configured incorrectly and cannot be used to import data.

### DISCUSSION THREAD

**Discussion by DSew20** [firefly-iii/firefly-iii#8019] [Firefly III] [answered]
Hello,

I thought I had Nordigen working.  Seems not.  I have a Nordigen account, but something seems wrong.  How do I edit the Nordigen API configuration data to correct issues?  I assume there is a file somewhere I need to update / delete?

I'm running Firefly III as a stack in Portainer (Docker), with 4 containers: cron, db, core (v6.0.26) and importer (v1.3.8).  I had this working 6 weeks ago, but can't get the importer to run.  All containers are updated.

Any pointers how to find / fix Nordigen API issues?

**Comment by JC5** [comment]
So how did you set it up the first time?

**Reply by DSew20** [reply]
Thank you.  I set up from the importer GUI - by clicking "Import from Nordigen".  But, the option is not available - just the message "The Nordigen API is configured incorrectly and cannot be used to import data.".

So, how do I reenter the Nordigen ID and Key?  There is no option to change this from the GUI.

**Reply by JC5** [reply]
There's a button Reauthenticate you can press. It also works to browse to /flush

**Comment by DSew20** [comment]
Thank you again.  I had tried the reauthenticate button before posting my original request.  I have tried again just now, and reentered my Client ID.  But, there is no Nordigen button.

I also tried a ctrl f5 and cleared the browser (Chrome) cache.  Still can't see the button to import from Nordigen.

**Comment by DSew20** [comment]
Does anybody have any ideas how to get the "Import from Nordigen" back or reset the ID and Key in a file / config?

@JC5 - Thank you.  I've tried the Reauthentication and clearing browser etc - no joy.

**Reply by JC5** [reply]
So what happens when you browse to /flush on your installation? That should totally reset it. 

Unless you set the Nordigen variables in the .env file or in Docker variables.

**Comment by DSew20** [comment]
Thanks again.  I brwosed to /flush.  Also tried Options>Admin>CleareCache (which I see does the same).  Cleared cookies, hard reload.  Still the same.  I get to the Nordigen settings / button.

I do see the button appear for a  fraction of a second on page refresh, but cant access / click.

**Reply by JC5** [reply]
So when i said "browse to /flush", could you try that on the data importer page too?

**Reply by DSew20** [reply]
Yes, I tried /flush on the importer too.  It takes me back to the Authenticate with Firefly III (/token) page.  I enter the Client ID.  But no options under Import from Nordigen.

**Reply by JC5** [reply]
So the Nordigen configuration variables, they're not in environment variables?

**Reply by DSew20** [reply]
The Nordigen ID and Key are in the env variables.

**Reply by JC5** [reply]
Seriously? Literally the first thing I asked you was how you set up nordigen. And this was your answer:

> Thank you. I set up from the importer GUI - by clicking "Import from Nordigen". But, the option is not available - just the message "The Nordigen API is configured incorrectly and cannot be used to import data.".

> So, how do I reenter the Nordigen ID and Key? There is no option to change this from the GUI.

So since you clearly did not use the GUI but used the environment variables, I can finally tell you that you can change the environment variables again.

**Reply by DSew20** [reply]
Thank you.  Sorry, to be clear, I did use the GUI originally when first setting up (2 months ago).  What I stated was correct.

Then based on your reply yesterday "Unless you set the Nordigen variables in the .env file or in Docker variables." - I read this as a suggestion to add ID and Key to the Environment Variables.  Which I did yesterday.  This did not fix the issue.

**Comment by DSew20** [comment, marked as answer]
I decided to go nuclear and managed to get Nordigen working.  I'm not sure if all of these steps are required, but this is what I did to get it running again:

- Browsed to /flush on FireFly and Importer
- Shut down the containers
- Cleared chrome cache
- Deleted the Nordigen cookies
- Set up a new Nordigen ID and Key
- Closed and reopened the Browser (Chrome)
- Restarted the containers
- Browsed to /flush on FireFly and Importer (again)
- Reathenticated
- I could now see the Import from Nordigen option, so set up in the GUI with the new ID and Key
- Ran an import
